ipfs/ipfs_test.go

```diff
@@ -0,0 +1,133 @@
+package ipfs
+
+import (
+	"context"
+	"log"
+	"testing"
+
+	status "github.com/sonr-io/core/errors"
+)
+
+func TestTempNode(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	/// --- Part I: Create an IPFS API node
+	node, err := SpawnEphemeral(ctx)
+	if err != nil {
+		t.Error(err)
+	}
+	if node == nil {
+		t.Errorf("SpawnDefault(ctx) resulted in nil result")
+		return
+	}
+
+	/// --- Part II: Upload a file to IPFS
+	data := []byte("Hello World!!!")
+	respUp, err := UploadData(ctx, data, node)
+	if err != nil {
+		t.Errorf("UploadData([]byte, coreAPI) resulted in status %d", respUp.Status)
+		t.Error(err)
+	}
+
+	/// --- Part III: Retrieve a file off of IPFS
+	bootstrapNodes := []string{
+		// IPFS Bootstrapper nodes.
+		"/dnsaddr/bootstrap.libp2p.io/p2p/QmNnooDu7bfjPFoTZYxMNLWUQJyrVwtbZg5gBMjTezGAJN",
+		"/dnsaddr/bootstrap.libp2p.io/p2p/QmQCU2EcMqAqQPR2i9bChDtGNJchTbq5TbXJJ16u19uLTa",
+		"/dnsaddr/bootstrap.libp2p.io/p2p/QmbLHAnMoJPWSCR5Zhtx6BHJX9KiKNN6tpvbUcqanj75Nb",
+		"/dnsaddr/bootstrap.libp2p.io/p2p/QmcZf59bWwK5XFi76CZX8cbJ4BhTzzA3gU1ZjYZcYW3dwt",
+
+		// IPFS Cluster Pinning nodes
+		"/ip4/138.201.67.219/tcp/4001/p2p/QmUd6zHcbkbcs7SMxwLs48qZVX3vpcM8errYS7xEczwRMA",
+		"/ip4/138.201.67.219/udp/4001/quic/p2p/QmUd6zHcbkbcs7SMxwLs48qZVX3vpcM8errYS7xEczwRMA",
+		"/ip4/138.201.67.220/tcp/4001/p2p/QmNSYxZAiJHeLdkBg38roksAR9So7Y5eojks1yjEcUtZ7i",
+		"/ip4/138.201.67.220/udp/4001/quic/p2p/QmNSYxZAiJHeLdkBg38roksAR9So7Y5eojks1yjEcUtZ7i",
+		"/ip4/138.201.68.74/tcp/4001/p2p/QmdnXwLrC8p1ueiq2Qya8joNvk3TVVDAut7PrikmZwubtR",
+		"/ip4/138.201.68.74/udp/4001/quic/p2p/QmdnXwLrC8p1ueiq2Qya8joNvk3TVVDAut7PrikmZwubtR",
+		"/ip4/94.130.135.167/tcp/4001/p2p/QmUEMvxS2e7iDrereVYc5SWPauXPyNwxcy9BXZrC1QTcHE",
+		"/ip4/94.130.135.167/udp/4001/quic/p2p/QmUEMvxS2e7iDrereVYc5SWPauXPyNwxcy9BXZrC1QTcHE",
+
+		// You can add more nodes here, for example, another IPFS node you might have running locally, mine was:
+		// "/ip4/127.0.0.1/tcp/4010/p2p/QmZp2fhDLxjYue2RiUvLwT9MWdnbDxam32qYFnGmxZDh5L",
+		// "/ip4/127.0.0.1/udp/4010/quic/p2p/QmZp2fhDLxjYue2RiUvLwT9MWdnbDxam32qYFnGmxZDh5L",
+	}
+	go func() {
+		err := ConnectToPeers(ctx, node, bootstrapNodes)
+		if err != nil {
+			log.Printf("failed connect to peers: %s", err)
+		}
+	}()
```

Comment:
How does this relate to the rest of the test? It's likely this will not be done before the download is attempted. Is this also needed for the upload?

Reply:
The CID below is pulled from these peers.

Reply:
If we wanted to uplload to this network then we would have to bootstrap before the upload I believe

Reply:
Okay makes sense. My only question now is: Is it possible that it will try to fetch the CID before this completes? That would make the test fail right?